ql/cashflows/overnightindexedcoupon.cpp

```diff
@@ -86,22 +88,24 @@ namespace QuantLib {
                 // forward part using telescopic property in order
                 // to avoid the evaluation of multiple forward fixings
                 if (i<n) {
-                    Handle<YieldTermStructure> curve =
-                        index->forwardingTermStructure();
+                    const Handle<YieldTermStructure> curve = index->forwardingTermStructure();
                     QL_REQUIRE(!curve.empty(),
-                               "null term structure set to this instance of "<<
-                               index->name());
+                               "null term structure set to this instance of " << index->name());
 
-                    const vector<Date>& dates = coupon_->valueDates();
-                    DiscountFactor startDiscount = curve->discount(dates[i]);
-                    DiscountFactor endDiscount = curve->discount(dates[n]);
+                    const DiscountFactor startDiscount = curve->discount(valueDates[i]);
+                    const DiscountFactor endDiscount = curve->discount(*endDateIterator);
 
-                    compoundFactor *= startDiscount/endDiscount;
+                    compoundFactor *= startDiscount / endDiscount;
                 }
 
-                Rate rate = (compoundFactor - 1.0) / coupon_->accrualPeriod();
+                const Rate rate = (compoundFactor - 1.0) / coupon_->accruedPeriod(date);
                 return coupon_->gearing() * rate + coupon_->spread();
             }
+            Rate swapletRate() const override {
+
+                const Date accrualEndDate = coupon_->valueDates()[coupon_->dt().size()];
+                return swapletRate(accrualEndDate);
```

Comment:
You already have `accrualEndDate_` as a data member, you can use that.

Reply:
> You already have `accrualEndDate_` as a data member, you can use that.

ok but this is in pricer, so I don't have access to that data member, but I can use its accessor.